qiskit_alice_bob_provider/processor/logical_cat.py

```diff
@@ -0,0 +1,247 @@
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+##############################################################################
+
+from typing import Dict, Iterator, List, Tuple
+
+import numpy as np
+
+from .description import (
+    AppliedInstruction,
+    InstructionProperties,
+    ProcessorDescription,
+)
+from .utils import (
+    compose_1q_errors,
+    full_flip_error,
+    pauli_errors_to_chi,
+    tensor_errors,
+)
+
+_1Q_INSTRUCTIONS = [
+    'x',
+    'z',
+    'p0',
+    'p1',
+    'p+',
+    'p-',
+    'mx',
+    'mz',
+    't',
+    'tdg',
+    'h',
+]
+
+
+class LogicalCatProcessor(ProcessorDescription):
+    """A description of a logical quantum processor whose logical qubits are
+    made of physical cat qubits.
+
+    The physical cat qubits are assembled into logical qubits using a linear
+    repetition code aiming to correct phase flips. Bit flips are already
+    addressed by the physical error correction built into the physical cat
+    qubits.
+
+    All logical qubits are assumed to have the same properties,
+    entirely controlled by the following quantities:
+    * distance (unitless), the distance of the repetition code used to correct
+      phase flips. This is also the number of physical cat qubits that form
+      a logical qubit
+    * kappa_1 in Hz, the one-photon dissipation rate of the memory of physical
+      cat qubits
+    * kappa_2 in Hz, the two-photon dissipation rate of the memory of physical
+      cat qubits
+    * alpha (unitless), the amplitude of the cat state or coherent state in the
+      memory of physical cat qubits. The average number of photons in the
+      memory is equal to |alpha|^2.
+
+    Note that on an actual chip, kappa_1 and kappa_2 depend on the architecture
+    and cannot be changed. The amplitude alpha however can be adjusted by the
+    operator of the chip.
+
+    Qubit connectivity is all-to-all.
+
+    The formulas for gate times and error models used in this description are
+    taken from different sources referenced in the code below.
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        n_qubits: int = 5,
+        distance: int = 11,
+        kappa_1: float = 100,
+        kappa_2: float = 10_000_000,
+        alpha: float = 4,
+        clock_cycle: float = 1e-9,
+    ):
+        if distance % 2 != 1 or distance < 3:
+            raise ValueError(
+                'The distance of the linear repetition code should be an odd '
+                'number >= 3'
+            )
+        if alpha < 2.0:
+            raise ValueError(
+                'The amplitude alpha should be a float larger than 2.0'
```

Comment:
People may be confused about this message as it looks unclear if the value 2.0 is allowed or not (also applies to the errors below and the previous commit).

The best wording could be Larger than or equal to, or simply >= as you do above. What do you think?

Reply:
I used >= like you suggested. Thanks!